astropy/table/index.py

```diff
@@ -975,22 +1011,58 @@ class TableLoc:
     ----------
     table : Table
         Indexed table to use
+    index_name : tuple or None
+        If not None, the index name as a tuple to use for all retrievals. If None
+        (default), the primary key index is used.
     """
 
-    def __init__(self, table):
+    def __init__(self, table: Table, index_name: tuple | None = None):
```

Comment:
supposedly ?
```suggestion
    def __init__(self, table: Table, index_name: str | tuple[str, ...] | None = None):
```

Reply:
Internally for this class, `index_id` must be `None` or `tuple[str, ...]`. I made the change.